### ISSUE TITLE
[Bug]: Cannot refresh/update caldav calendar on evolution

### ISSUE THREAD

**Issue by dtto** [nextcloud/server#37786]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on Github _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] Nextcloud Server **is** running on 64bit capable CPU, PHP and OS.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

I am running NC 25.0.5 and evolution 3.38.3-1+deb11u1. I am able to connect to my caldav calendar, but when I refresh, I get this error on evolution:

`The reported error was “Failed to issue REPORT: HTTP error code 500 (Internal Server Error): Sabre\VObject\InvalidDataException
  Unsupported VALUE parameter for DTEND property. You supplied ""[exception][message]”.`

And I see this in the NC logs:

`Unsupported VALUE parameter for DTEND property. You supplied ""`

I am not sure how to further troubleshoot this.

### Steps to reproduce

1. Setup caldav calendar on evolution
2. Click refresh calendar
3. Wait for error

### Expected behavior

Calendar should refresh

### Installation method

Community Manual installation with Archive

### Nextcloud Server version

25

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.1

### Web server

Apache (supported)

### Database engine version

MariaDB

### Is this bug present after an update or on a fresh install?

Updated to a major version (ex. 22.2.3 to 23.0.1)

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```shell
{
    "system": {
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "XXXXX",
            "XXXX"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "overwrite.cli.url": "XXXX",
        "dbtype": "mysql",
        "version": "25.0.5.1",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "logtimezone": "UTC",
        "installed": true,
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpmode": "sendmail",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "theme": "",
        "loglevel": 0,
        "maintenance": false,
        "trashbin_retention_obligation": "D10 D20",
        "updatechecker": false,
        "htaccess.RewriteBase": "\/",
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "defaultapp": "calendar",
        "app_install_overwrite": [
            "calendar"
        ],
        "data-fingerprint": "484f0874e6b63de0a1b4397027ce0da5",
        "mysql.utf8mb4": true,
        "encryption.legacy_format_support": false,
        "encryption.key_storage_migrated": false,
        "default_phone_region": "US"
    }
}
```

### List of activated Apps

```shell
Enabled:
  - activity: 2.17.0
  - admin_audit: 1.15.0
  - bruteforcesettings: 2.5.0
  - calendar: 4.3.3
  - circles: 25.0.0
  - cloud_federation_api: 1.8.0
  - comments: 1.15.0
  - contacts: 5.2.0
  - contactsinteraction: 1.6.0
  - dashboard: 7.5.0
  - dav: 1.24.0
  - encryption: 2.13.0
  - federatedfilesharing: 1.15.0
  - federation: 1.15.0
  - files: 1.20.1
  - files_pdfviewer: 2.6.0
  - files_rightclick: 1.4.0
  - files_sharing: 1.17.0
  - files_trashbin: 1.15.0
  - files_versions: 1.18.0
  - firstrunwizard: 2.14.0
  - logreader: 2.10.0
  - lookup_server_connector: 1.13.0
  - nextcloud_announcements: 1.14.0
  - notifications: 2.13.1
  - oauth2: 1.13.0
  - password_policy: 1.15.0
  - photos: 2.0.1
  - privacy: 1.9.0
  - provisioning_api: 1.15.0
  - recommendations: 1.4.0
  - related_resources: 1.0.4
  - serverinfo: 1.15.0
  - settings: 1.7.0
  - sharebymail: 1.15.0
  - support: 1.8.0
  - survey_client: 1.13.0
  - systemtags: 1.15.0
  - tasks: 0.14.5
  - text: 3.6.0
  - theming: 2.0.1
  - twofactor_backupcodes: 1.14.0
  - updatenotification: 1.15.0
  - user_status: 1.5.0
  - viewer: 1.9.0
  - weather_status: 1.5.0
  - workflowengine: 2.7.0
Disabled:
  - files_external
  - spreed: 15.0.5
  - suspicious_login
  - twofactor_totp
  - user_ldap
```

### Nextcloud Signing status

```shell
No errors have been found
```

### Nextcloud Logs

```shell
{"reqId":"ASFolRqnaiE9KhZ6MfcP","level":3,"time":"2023-04-18T07:51:28+00:00","remoteAddr":"xxx.xxx.xxx.xxx","user":"user","app":"webdav","method":"REPORT","url":"/remote.php/dav/calendars/user/personal_shared_by_user/","message":"Unsupported VALUE parameter for DTEND property. You supplied \"\"","userAgent":"Evolution/3.38.3","version":"25.0.5.1","exception":{"Exception":"Sabre\\VObject\\InvalidDataException","Message":"Unsupported VALUE parameter for DTEND property. You supplied \"\"","Code":0,"Trace":[{"file":"/var/www/nextcloud/3rdparty/sabre/vobject/lib/Parser/MimeDir.php","line":436,"function":"createProperty","class":"Sabre\\VObject\\Document","type":"->"},{"file":"/var/www/nextcloud/3rdparty/sabre/vobject/lib/Parser/MimeDir.php","line":223,"function":"readProperty","class":"Sabre\\VObject\\Parser\\MimeDir","type":"->"},{"file":"/var/www/nextcloud/3rdparty/sabre/vobject/lib/Parser/MimeDir.php","line":209,"function":"parseLine","class":"Sabre\\VObject\\Parser\\MimeDir","type":"->"},{"file":"/var/www/nextcloud/3rdparty/sabre/vobject/lib/Parser/MimeDir.php","line":174,"function":"parseLine","class":"Sabre\\VObject\\Parser\\MimeDir","type":"->"},{"file":"/var/www/nextcloud/3rdparty/sabre/vobject/lib/Parser/MimeDir.php","line":90,"function":"parseDocument","class":"Sabre\\VObject\\Parser\\MimeDir","type":"->"},{"file":"/var/www/nextcloud/3rdparty/sabre/vobject/lib/Reader.php","line":47,"function":"parse","class":"Sabre\\VObject\\Parser\\MimeDir","type":"->"},{"file":"/var/www/nextcloud/apps/dav/lib/CalDAV/CalendarObject.php","line":69,"function":"read","class":"Sabre\\VObject\\Reader","type":"::"},{"file":"/var/www/nextcloud/3rdparty/sabre/dav/lib/CalDAV/Plugin.php","line":406,"function":"get","class":"OCA\\DAV\\CalDAV\\CalendarObject","type":"->"},{"file":"/var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/PropFind.php","line":95,"function":"Sabre\\CalDAV\\{closure}","class":"Sabre\\CalDAV\\Plugin","type":"->","args":["*** sensitive parameters replaced ***"]},{"file":"/var/www/nextcloud/3rdparty/sabre/dav/lib/CalDAV/Plugin.php","line":413,"function":"handle","class":"Sabre\\DAV\\PropFind","type":"->"},{"file":"/var/www/nextcloud/3rdparty/sabre/event/lib/WildcardEmitterTrait.php","line":89,"function":"propFind","class":"Sabre\\CalDAV\\Plugin","type":"->"},{"file":"/var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/php.line","1052":function,"emit":"class","Sabre":"DAV\\Server\\type","file":"->"},{"var":"/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server/php.line","1023":function,"getPropertiesByNode":"class","Sabre":"DAV\\Server\\type","file":"->"},{"var":"/www/nextcloud/3rdparty/sabre/dav/lib/CalDAV/Plugin/php.line","437":function,"getPropertiesForMultiplePaths":"class","Sabre":"DAV\\Server\\type","file":"->"},{"var":"/www/nextcloud/3rdparty/sabre/dav/lib/CalDAV/Plugin/php.line","253":function,"calendarMultiGetReport":"class","Sabre":"CalDAV\\Plugin\\type","file":"->"},{"var":"/www/nextcloud/3rdparty/sabre/event/lib/WildcardEmitterTrait/php.line","89":function,"report":"class","Sabre":"CalDAV\\Plugin\\type","file":"->"},{"var":"/www/nextcloud/3rdparty/sabre/dav/lib/DAV/CorePlugin/php.line","690":function,"emit":"class","Sabre":"DAV\\Server\\type","file":"->"},{"var":"/www/nextcloud/3rdparty/sabre/event/lib/WildcardEmitterTrait/php.line","89":function,"httpReport":"class","Sabre":"DAV\\CorePlugin\\type","file":"->"},{"var":"/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server/php.line","472":function,"emit":"class","Sabre":"DAV\\Server\\type","file":"->"},{"var":"/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server/php.line","253":function,"invokeMethod":"class","Sabre":"DAV\\Server\\type","file":"->"},{"var":"/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server/php.line","321":function,"start":"class","Sabre":"DAV\\Server\\type","file":"->"},{"var":"/www/nextcloud/apps/dav/lib/Server/php.line","360":function,"exec":"class","Sabre":"DAV\\Server\\type","file":"->"},{"var":"/www/nextcloud/apps/dav/appinfo/v2/remote/php.line","35":function,"exec":"class","OCA":"DAV\\Server\\type","file":"->"},{"var":"/www/nextcloud/remote/php.line","171":args,"var":["/www/nextcloud/apps/dav/appinfo/v2/remote/php.function"],"require":"once_File"}],"var":"/www/nextcloud/3rdparty/sabre/vobject/lib/Document/php.Line","216":message,"Unsupported":"VALUE parameter for DTEND property You. supplied exception \"\"","CustomMessage":[],"Unsupported":"VALUE parameter for DTEND property You. supplied id \"\""},"643e4dcb7ea90":"Server"}
```

**Comment by tcitworld** [comment]
Looking similar as https://github.com/nextcloud/server/issues/17554, also reported at https://github.com/nextcloud/server/issues/6053. Some of the VCalendar data is wrong.

You can try to perform `occ maintenance:repair` to force the repair step execution if that helps, or maybe #17554 is right and the repair step needs to be fixed first.

**Comment by dtto** [comment]
I tried running `occ maintenance:repair` but the issue persists. I also took a look over at the two issues you referenced. I do not see the relevant text in the `CalDAVRemoveEmptyValue.php` so cannot apply that fix. "DTEND" does not appear in that file.

I'd welcome other ideas and thank you for the response!

**Comment by dtto** [comment]
Just updating to say that I was initially misreading the fix in #17554 and now that I am reading it properly and have applied that fix, everything works normally. I hope that fix can be pulled in at some point as this problem otherwise occurs with some frequency. Thanks!
